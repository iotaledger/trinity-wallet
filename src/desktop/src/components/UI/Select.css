.select {
    width: 100%;
    margin-bottom: 50px;
    select {
        font-family: 'Lato';
        font-weight: 200;
        font-size: var(--font-size-default);
        line-height: 26px;
        width: 100%;
        border: none;
        background: #2a4a51;
        color: #fff;
        padding: 6px 12px;
        resize: none;
        -webkit-appearance: none;
        -webkit-border-radius: 0px;
        border-radius: 4px;
        cursor: pointer;
        &:focus {
            outline: none;
        }
    }
    > div {
        position: relative;
        &:after {
            content: '';
            display: block;
            width: 0;
            height: 0;
            border-style: solid;
            border-width: 6px 4px 0 4px;
            border-color: #fff transparent transparent transparent;
            position: absolute;
            right: 12px;
            top: calc(50% - 2px);
            opacity: 0.6;
            pointer-event: none;
        }
        &:hover:after {
            opacity: 1;
        }
    }
}
