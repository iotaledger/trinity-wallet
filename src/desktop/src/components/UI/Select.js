import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import css from './Select.css';

export default class Select extends React.PureComponent {
    static propTypes = {
        children: PropTypes.node,
        label: PropTypes.string,
        variant: PropTypes.oneOf(['default', 'success', 'warning']).isRequired
    };

    static defaultProps = {
        variant: 'default'
    };

    render() {
        const { children, variant, label } = this.props;

        return (
            <div className={classNames(css.select, css[variant])}>
                {label ? <label>{label}</label> : null}
                <div>
                    <select {...this.props}>{children}</select>
                </div>
            </div>
        );
    }
}
