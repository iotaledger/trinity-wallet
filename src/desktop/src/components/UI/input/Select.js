import React from 'react';
import PropTypes from 'prop-types';
import css from './Input.css';

export default class Select extends React.PureComponent {
    static propTypes = {
        children: PropTypes.node,
        label: PropTypes.string,
    };

    render() {
        const { children, label } = this.props;

        return (
            <div className={css.input}>
                <fieldset>
                    <a className={css.down}>&shy;</a>
                    <select {...this.props}>{children}</select>
                    <small>{label}</small>
                </fieldset>
            </div>
        );
    }
}
