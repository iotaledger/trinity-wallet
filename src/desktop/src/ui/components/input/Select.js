import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import Scrollbar from 'ui/components/Scrollbar';

import css from './input.scss';

/**
 * Select component
 */
export default class Select extends React.PureComponent {
    static propTypes = {
        /** Current selected value */
        value: PropTypes.string.isRequired,
        /** Select dropdown options */
        options: PropTypes.array.isRequired,
        /** Select item label */
        label: PropTypes.string.isRequired,
        /** Is the component disabled */
        disabled: PropTypes.bool,
        /** Select event callback
         * @param {Bool} value - Current checkbox state
         * @returns {Void}
         */
        onChange: PropTypes.func.isRequired,
    };

    constructor(props) {
        super(props);
        this.state = {
            open: false,
        };
        this.onClick = this.clickOutside.bind(this);
    }

    componentDidMount() {
        window.addEventListener('click', this.onClick);
    }

    componentWillUnmount() {
        window.removeEventListener('click', this.onClick);
    }

    clickOutside(e) {
        if (this.state.open) {
            const area = this.select;
            if (!area.contains(e.target)) {
                this.setState({
                    open: false,
                });
            }
        }
    }

    choose = (value) => {
        this.setState({
            open: false,
        });
        this.props.onChange(value);
    };

    render() {
        const { value, options, label, disabled } = this.props;
        const { open } = this.state;

        return (
            <div className={classNames(css.input, css.select, open ? css.open : null, disabled ? css.disabled : null)}>
                <fieldset
                    ref={(el) => {
                        this.select = el;
                    }}
                >
                    <div
                        onClick={() =>
                            this.setState({
                                open: !open,
                            })
                        }
                        className={css.selectable}
                    >
                        {value}
                        <small>&shy;</small>
                    </div>
                    {open && (
                        <ul>
                            <Scrollbar>
                                {options.map((option) => (
                                    <li
                                        className={option.label === value ? css.selected : null}
                                        key={option.value}
                                        onClick={() => this.choose(option.value)}
                                    >
                                        {option.label}
                                    </li>
                                ))}
                            </Scrollbar>
                        </ul>
                    )}
                    <small>{label}</small>
                </fieldset>
            </div>
        );
    }
}
