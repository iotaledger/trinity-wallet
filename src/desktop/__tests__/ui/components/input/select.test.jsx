import React from 'react';
import { shallow } from 'enzyme';

import Select from 'ui/components/input/Select';

const props = {
    value: 'Foo',
    valueLabel: 'Foo Bar',
    options: [
        {
            value: 'Foo',
            label: 'Foo Bar',
        },
        {
            value: 'Fizz',
            label: 'Fizz Buzz',
        },
    ],
    label: 'Lorem Ipsum',
    onChange: jest.fn(),
};

describe('Select component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Select {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Input value', () => {
        const wrapper = shallow(<Select {...props} />);

        expect(wrapper.find('.selectable').text()).toEqual('Foo Bar');
    });

    test('Input change callback', () => {
        const wrapper = shallow(<Select {...props} />);

        wrapper.setState({ open: true });

        wrapper
            .find('li')
            .last()
            .simulate('click');

        expect(props.onChange).toHaveBeenLastCalledWith('Fizz');
    });

    test('Input disabled state', () => {
        const mockProps = Object.assign({}, props, { disabled: true });
        const wrapper = shallow(<Select {...mockProps} />);

        expect(wrapper.hasClass('disabled')).toBeTruthy();
    });
});
