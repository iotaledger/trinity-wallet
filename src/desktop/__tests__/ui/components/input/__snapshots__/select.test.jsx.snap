// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Render the component 1`] = `
<div
  className="input select"
>
  <fieldset>
    <div
      className="selectable"
      onClick={[Function]}
    >
      Foo Bar
    </div>
    <small>
      Lorem Ipsum
    </small>
  </fieldset>
</div>
`;
